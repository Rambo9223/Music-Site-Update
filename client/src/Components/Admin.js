import backgroundAdmin from "../Images/Live-Wide.jpg";
import { useMediaQuery } from "react-responsive";
import Login from "./Login";

// Admin portal page for managing website
export default function Admin(){

    // useMediaQueries for responsivness 
    const isDesktopOrLaptop = useMediaQuery({ query: "(min-width: 1024px)" });
    const isLaptop = useMediaQuery({ query: "(max-width: 1023px)" });
    const isTablet = useMediaQuery({ query: "(min-width: 651px)" });
    const isMobile = useMediaQuery({ query: "(max-width: 650px)" });
 

  const content = 
  <div id="contact-content">
  
  {/*We display the login button/form until the admin is logged in*/}
  {isLaptop?<div className="break-large"></div>:null}
  <Login/>
  </div>;

    return (
        <>
        {isDesktopOrLaptop ? (
          <>
            <div id="background-contact">
              <img id="contact-image" src={backgroundAdmin} alt="" />
            </div>
            </>
        ) : null}
        {isLaptop ? (
          <>
            <div id="background-contact-small">
              <img id="contact-image" src={backgroundAdmin} alt="" />
            </div>
            {isTablet ? <div className="break"></div> : null}
            {isMobile ? <div className="break-xl" /> : null}
            </>
        ) : null}
            {content}
            <div className="break-small" /><div className="break-small" />          
      </>
        
    )
}