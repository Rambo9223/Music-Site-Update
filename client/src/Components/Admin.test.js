/**
*@jest-environment jsdom 
*/
import { render,cleanup, waitFor} from '@testing-library/react';
import * as React from 'react';
import Admin from './Admin';

afterEach(()=>{
  cleanup();
});

describe('Admin Page Tests', () => { 

test('Page matches snapshot', () => { 
    let page = render(<Admin/>);
    expect(page).toMatchSnapshot();
 })

test('Login Component is visible', async () => {
    let page = render(<Admin/>);

    let loginButton = await (page.findByText("Login"));

    expect(loginButton).toBeInTheDocument();
});

});
