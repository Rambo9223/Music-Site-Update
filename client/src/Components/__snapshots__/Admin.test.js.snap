// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page Tests Page matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="contact-content"
      >
        <div>
          <button
            class="rounded-pill btn btn-primary"
            id="login-button"
            type="button"
          >
            Login
          </button>
        </div>
      </div>
      <div
        class="break-small"
      />
      <div
        class="break-small"
      />
    </div>
  </body>,
  "container": <div>
    <div
      id="contact-content"
    >
      <div>
        <button
          class="rounded-pill btn btn-primary"
          id="login-button"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
    <div
      class="break-small"
    />
    <div
      class="break-small"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
